core/src/components/range/range.tsx

```diff
@@ -213,7 +213,12 @@ export class Range implements ComponentInterface {
    */
   @Prop({ mutable: true }) value: RangeValue = 0;
   @Watch('value')
-  protected valueChanged() {
+  protected valueChanged(newVlue: RangeValue, oldValue: RangeValue) {
+    // The check is necessary because the value can be an object.
+    if (JSON.stringify(newVlue) !== JSON.stringify(oldValue)) {
```

Comment:
JSON.stringify is a bit slow. I don't know that it super matters in the long run, but it would be significantly more efficient if you compared the properties without stringifying them. I know it's a bit annoying because they're a union type of a number and object, but worth considering I guess.

Reply:
[Done](https://github.com/ionic-team/ionic-framework/pull/30293/commits/e8b57c838b5c7d9658f86530a886c16236fd1c7e)!